pkg-py/src/querychat/_querychat.py

```diff
@@ -241,6 +275,170 @@ def generate_greeting(self, *, echo: Literal["none", "output"] = "none"):
         client.set_turns([])
         return str(client.chat(GREETING_PROMPT, echo=echo))
 
+    def _assemble_system_prompt(
```

Comment:
Did you intend to also get rid of `def assemble_system_prompt`? 

BTW, I feel like this doesn't necessarily have to be a method: it could take just `tools`, `data_source`, and `prompt_parts` as input params.